# Prefetch with certain cache keys after the hashtag of each URLs.
diff --git a/node_modules/expo-image/.DS_Store b/node_modules/expo-image/.DS_Store
new file mode 100644
index 0000000..0d4eecc
Binary files /dev/null and b/node_modules/expo-image/.DS_Store differ
diff --git a/node_modules/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt b/node_modules/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt
index a3fae8a..cf1b5a9 100644
--- a/node_modules/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt
+++ b/node_modules/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt
@@ -25,10 +25,18 @@ class ExpoImageModule : Module() {
     Function("prefetch") { urls: List<String> ->
       val context = appContext.reactContext ?: return@Function
       urls.forEach {
-        Glide
-          .with(context)
-          .download(GlideUrl(it))
-          .submit()
+        var separated = it.split("#")
+        if (separated.size > 1) {
+          Glide
+            .with(context)
+            .download(GlideUrlWithCustomCacheKey(it, null, it.split("#")[1]))
+            .submit()
+        } else {
+          Glide
+            .with(context)
+            .download(GlideUrl(it))
+            .submit()
+        }
       }
     }
 
diff --git a/node_modules/expo-image/ios/ImageModule.swift b/node_modules/expo-image/ios/ImageModule.swift
index 04f2dae..7c7bb7d 100644
--- a/node_modules/expo-image/ios/ImageModule.swift
+++ b/node_modules/expo-image/ios/ImageModule.swift
@@ -91,7 +91,13 @@ public final class ImageModule: Module {
     }
 
     Function("prefetch") { (urls: [URL]) in
-      SDWebImagePrefetcher.shared.prefetchURLs(urls)
+      SDWebImagePrefetcher.shared.prefetchURLs(urls, context: [.cacheKeyFilter : SDWebImageCacheKeyFilter { url in
+            var components = url.absoluteString.components(separatedBy: "#")
+            guard components.count > 1 else {
+                return url.absoluteString
+            }
+            return url.absoluteString.components(separatedBy: "#")[1]
+          }], progress: .none)
     }
 
     AsyncFunction("clearMemoryCache") { () -> Bool in
